spec/mongoid/traversable_spec.rb

```diff
@@ -904,10 +906,9 @@ class GlobalDuplicateDiscriminatorKeyChild < GlobalDuplicateDiscriminatorKeyPare
     end
 
     context "when the global discriminator key conflicts with mongoid's internals" do
-
-      after do
-        Mongoid.discriminator_key = "_type"
-      end
+      # This is not an override, it is meant to restore the original value
+      # after the test only.
+      config_override :discriminator_key, '_type'
```

Comment:
Why is this an improvement over the other code? Seems like just setting it afterwards is better since it's the intended outcome.

Reply:
ah